### ISSUE TITLE
Entries marked as paid, but no PayPal funds sent?

### ISSUE THREAD

**Issue by printer1975** [geoffhumphrey/brewcompetitiononlineentry#608]
Is there another way that an entrant can get their entries marked as paid without paying through Paypal, or me checking it off in the system? I had two register this weekend, and both are marked as paid.

**Comment by geoffhumphrey** [comment]
No, there isn't a readily apparent way to Joe Schmo user. There is one way buried deep in the code that I know of, but I won't specify it here for obvious reasons.

**Comment by printer1975** [comment]
Something is going on then. All registered entries received 5/14, 5/15, & 5/16 are showing as paid, but there have been no subsequent payments thru PayPal...

**Comment by geoffhumphrey** [comment]
Have you recently upgraded to 2.1.1? If so, there's a snippet of legacy test code in the brew.sec.php file that will cause this issue, starting at line 240:
```
<?php 
if ($total_to_pay == 0) $brewPaid = "1";
else $brewPaid = $row_log['brewPaid'];
?>
```
Erasing this snippet should take care of the issue. It has also been removed in the latest commit.

**Comment by printer1975** [comment]
I didn't touch anything....we had that issue Friday night that you assisted in restoring the DB table that was corrupted. Could it have happened then?

**Comment by geoffhumphrey** [comment]
Yes, apologies, I didn't realize you were associated with a hosted account. All hosted accounts were updated. I'll make sure to update with the latest commit today.

**Comment by printer1975** [comment]
OK, cool, thanks!

**Comment by ercousin** [comment]
I see several changes to this file over the last few days.

What is the plan for 2.1.2? I already disabled the php warnings, but I'm a bit confused about what I need to change to avoid this issue...

![image](https://cloud.githubusercontent.com/assets/1551339/15404960/e72bd986-1dcd-11e6-8788-7bda5f8e5106.png)

**Comment by ercousin** [comment]
Also, in my sections/brew.sec.php, that php section shows up at line 481, not 240. Are there different versions of 2.1.1?

![image](https://cloud.githubusercontent.com/assets/1551339/15405205/eb4153f6-1dce-11e6-9bfc-91e4f808b657.png)

**Comment by geoffhumphrey** [comment]
Erase these lines altogether to avoid the entries being marked as paid erroneously. In the 2.1.1 release file, they are located beginning at 240. Not sure what version of the file you've got, @ercousin - maybe from a commit version? 
```
<?php 
if ($total_to_pay == 0) $brewPaid = "1";
else $brewPaid = $row_log['brewPaid'];
?>
<input type="hidden" name="brewPaid" value="<?php echo $brewPaid; ?>">
```
In any case, if you download the latest commit and replace those files on your server, things will be good. All hosted accounts have the fix in place. 

For 2.1.2, these lines will be stripped out. The original intention was to mark any entries as paid if a user has exceeded the entry fee cap (if enabled); however, the logic wasn't right. Instead, there will be a check AFTER the user submits the entry and is processed by the system.

**Comment by ercousin** [comment]
I now see it at line 240, must have just been a quirk of notepad++...

Thanks for the follow up. I have 2.1.1 installed with fixes to paths.php (for php notices) and brew.sec.php (for marking paid). Are there any other fixes required for 2.1.1 to get it working?

**Comment by geoffhumphrey** [comment]
If you've implemented the fixes detailed in issues #607, #609, #610, and #611 as well as this one, you should be good to go (unless other bug reports come in, of course). 2.1.2 is on the way. Target release is May 31.
